.github/workflows/e2e-tests.yml

```
@@ -16,33 +16,44 @@ permissions: {}
 
 jobs:
     e2e-tests:
-        name: 'E2E Tests (${{ matrix.wp }}) (${{ matrix.shard }})'
-        runs-on: ubuntu-latest
+        name: 'E2E Tests (${{ matrix.wp }}) (${{ matrix.project }}) (${{ matrix.shard }})'
+        # Temporarily use ARM macOS for WebKit tests to work around https://github.com/swissspidy/media-experiments/issues/611
+        runs-on: ${{ matrix.project == 'webkit' && 'macos-latest' || 'ubuntu-latest' }}
         continue-on-error: ${{ matrix.experimental == true }}
         strategy:
             fail-fast: false
             matrix:
                 wp: ['latest']
-                shard: ['1/4', '2/4', '3/4', '4/4']
+                project: ['chromium', 'firefox', 'webkit']
+                shard: ['1/2', '2/2']
                 experimental: [false]
                 include:
                     - wp: 'trunk'
-                      shard: '1/4'
+                      shard: '1/2'
+                      project: 'chromium'
                       experimental: true
                     - wp: 'trunk'
-                      shard: '2/4'
-                      experimental: true
-                    - wp: 'trunk'
-                      shard: '3/4'
-                      experimental: true
-                    - wp: 'trunk'
-                      shard: '4/4'
+                      shard: '2/2'
+                      project: 'chromium'
                       experimental: true
 
         steps:
             - name: Checkout
               uses: actions/checkout@v4
 
+            - name: Install Docker (missing on macOS)
+              if: runner.os == 'macOS'
+              run: |
+                brew install docker docker-compose colima
+
+                # See https://github.com/abiosoft/colima/discussions/874
+                mkdir -p ~/.docker/cli-plugins
+                ln -sfn $(brew --prefix docker-compose)/bin/docker-compose ~/.docker/cli-plugins/docker-compose
+
+                # See https://github.com/abiosoft/colima/issues/970#issuecomment-2224011277
+                sudo curl -L -o /opt/homebrew/Cellar/lima/0.22.0/bin/limactl https://github.com/mikekazakov/lima-nohvf/raw/master/limactl && sudo chmod +x /opt/homebrew/Cellar/lima/0.22.0/bin/limactl
+                colima start --network-address --arch arm64 --vm-type=qemu
```

Looking at the recent commits of Colima (e.g commit https://github.com/abiosoft/colima/commit/aa4d3e7780496e069fea43facff0f55b4bc67f34), I think one could just use `--vm-type=vz` out-of-the-box.
```suggestion
                colima start --network-address --arch arm64 --vm-type=vz --cpu 3 --memory 7
```
(I also added `--cpu 3 --memory 7` to match [the hardware used by M1 macOS runner](https://github.blog/changelog/2024-01-30-github-actions-introducing-the-new-m1-macos-runner-available-to-open-source/))

If that doesn't work, it should probably run under Rosetta 2 emulation.
```suggestion
                colima start --network-address --arch arm64 --vm-type=vz --vz-rosetta --cpu 3 --memory 7
```
https://github.com/abiosoft/colima/blob/73732ae8e44148b6191406f61044eff8195ae29b/README.md?plain=1#L152-L156

---

Thank you so much for helping me debug this!

If we can't get it to work, I'm exploring a Docker-less alternative in #614 to make things a bit easier.

---

Neither seems to work 😞 🤔

---

Hmm, it looks like `--vm-type=vz` also uses nested virtualization. :( 
```
{"level":"fatal","msg":"Error Domain=VZErrorDomain Code=2 Description=\"Invalid virtual machine configuration. Virtualization is not available on this hardware.\" UserInfo={\n    NSLocalizedFailure = \"Invalid virtual machine configuration.\";\n    NSLocalizedFailureReason = \"Virtualization is not available on this hardware.\";\n}","time":"2024-08-14T16:08:38Z"}
````
https://github.com/kleisauke/media-experiments/actions/runs/10391084038/job/28773154085#step:3:214

> I'm exploring a Docker-less alternative in #614 to make things a bit easier.

Ah, that would be neater. FWIW, Podman isn't supported on the `macos-14` runner either:
https://github.com/actions/runner-images/discussions/9304